From b3b6f892bfdf09cac54848110b269b9ec5f941a4 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Mon, 25 Feb 2019 19:24:35 +0200
Subject: [PATCH] p7zip: Silence all warnings

* There are too many warnings to fix, so just silence them
  to avoid build spam and hope upstream will fix them sometime

Change-Id: Ie2cf34e8c4e96fad09f8dd1110102206c50f4769
---
 Android.mk | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index b147ebd..c05b093 100644
--- a/Android.mk
+++ b/Android.mk
@@ -16,8 +16,25 @@ common_cflags := \
 common_cflags += \
     -fexceptions \
     -Wno-c++11-narrowing \
-    -Wno-error \
-    -Wno-non-virtual-dtor
+    -Wno-dangling-else \
+    -Wno-delete-non-virtual-dtor \
+    -Wno-deprecated-increment-bool \
+    -Wno-ignored-qualifiers \
+    -Wno-implicit-exception-spec-mismatch \
+    -Wno-logical-op-parentheses \
+    -Wno-missing-field-initializers \
+    -Wno-non-virtual-dtor \
+    -Wno-overloaded-virtual \
+    -Wno-reorder \
+    -Wno-self-assign \
+    -Wno-sign-compare \
+    -Wno-switch \
+    -Wno-unneeded-internal-declaration \
+    -Wno-unused-const-variable \
+    -Wno-unused-function \
+    -Wno-unused-parameter \
+    -Wno-unused-private-field \
+    -Wno-unused-variable
 
 common_c_includes := \
     $(LOCAL_PATH)/CPP \
-- 
2.17.1

